shared/subspace-data-retrieval/src/piece_fetcher.rs

```diff
@@ -13,7 +13,7 @@ use tracing::{debug, trace};
 // This code was copied and modified from subspace_service::sync_from_dsn::download_and_reconstruct_blocks():
 // <https://github.com/autonomys/subspace/blob/d71ca47e45e1b53cd2e472413caa23472a91cd74/crates/subspace-service/src/sync_from_dsn/import_blocks.rs#L236-L322>
 pub async fn download_pieces<PG>(
-    piece_indexes: &Vec<PieceIndex>,
+    piece_indexes: &[PieceIndex],
```

Comment:
if we want to clone anyway, wouldn't it be better to send the copy here instead ?